viewing_party/party.py

```diff
@@ -0,0 +1,344 @@
+    result_set=friends_set.difference(user_set)
+    
+    final_list=[]
+    for title in result_set:
+        final_dict={}
+        final_dict["title"]=title
+  
+        final_list.append(final_dict)
+    
+    return final_list        
+
+# *_____________________________ with many lists =S  __________________________________* 
+    #     user_watched=[]
+    # friends_watched=[]
+    # friends_unique_watched=[]
+    # for friend in user_data["friends"]:
+    #     for movie in friend["watched"]:
+    #         if movie not in friends_watched:
+    #             friends_watched.append(movie)
+            
+    # if len(user_data["watched"])>0:
+    #     for movie in user_data["watched"]:
+    #         if movie not in user_watched:
+    #             user_watched.append(movie)
+        
+    #     # for friend in user_data["friends"]:
+    #     #     for movie in friend["watched"]:
+    #     #         if movie not in friends_watched:
+    #     #             friends_watched.append(movie)  
+    #     for movie in friends_watched:
+    #         if movie not in user_watched:
+    #             friends_unique_watched.append(movie)
+    #             return friends_unique_watched
+    # return friends_watched
+    
+        
+        # return friends_unique_watched
+
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+
+
+def get_available_recs(user_data):
+  
+    recommendations=[]
+    friends_unique_watched=[]
+    
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_unique_watched:
+                friends_unique_watched.append(movie)
+  
+    # why it's not considering the user watched????
+    for movie in friends_unique_watched:
+        # if movie not in user_data["watched"]:
+       
+        if {"title":movie["title"]} not in user_data["watched"]:
+            if movie["host"] in user_data["subscriptions"]:
+                recommendations.append(movie)
+    return recommendations
+   
+
+
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+# *************************************----------------------------*******************************************
+
+def get_new_rec_by_genre(user_data):
+    genre_recommended=[]
+    friends_watched=[]
+    user_watched_genre=[]
+    
+    for movie in user_data["watched"]:
+        user_watched_genre.append(movie["genre"])
+    if len(user_watched_genre)>0:   
+        user_fav_genre=max(set(user_watched_genre), key = user_watched_genre.count)
```

Comment:
This is doing the same work as `get_most_watched_genre`, I recommend using that as a helper function here.